akka-remote/src/main/scala/akka/remote/artery/compress/HeavyHitters.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.remote.artery.compress
+
+import scala.collection.immutable
+
+// TODO describe / think about how we could handle an actor dying and removing from "top"
+// TODO this is the scenario when "super traffic actor" goes away and we want to not use up the slot in HH
```

Comment:
And that automatically removes no longer referenced actors. Also, this works in the weird case if an actor is dead here, but someone still keeps sending tons of remote messsages to it. Even though those messages are dropped *here* they still travel through the wire so it still makes sense to compress them.